artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompSession.java

```diff
@@ -389,7 +389,7 @@ public void sendInternalLarge(CoreMessage message, boolean direct) throws Except
       long id = storageManager.generateID();
       LargeServerMessage largeMessage = storageManager.createLargeMessage(id, message);
 
-      byte[] bytes = new byte[message.getBodyBuffer().writerIndex() - CoreMessage.BODY_OFFSET];
+      byte[] bytes = new byte[message.getBodyBuffer().readableBytes()];
```

Comment:
it would be safer to do it this way.. I'm sure it wouldn't affect other messages.. 
message.getBodyBuffer() may affect the position of a message if it's on a topic with many subscriptions.